crates/common/src/local_db/pipeline/adapters/bootstrap.rs

```diff
@@ -113,16 +114,30 @@ pub trait BootstrapPipeline {
         Ok(())
     }
 
-    async fn run<DB>(
+    async fn clear_orderbook_data<DB>(
         &self,
-        _db: &DB,
-        _db_schema_version: Option<u32>,
-        _config: &BootstrapConfig,
+        db: &DB,
+        ob_id: &OrderbookIdentifier,
     ) -> Result<(), LocalDbError>
     where
         DB: LocalDbQueryExecutor + ?Sized,
     {
-        Err(LocalDbError::MissingBootstrapImplementation)
+        db.query_text(&clear_orderbook_data_stmt(ob_id)).await?;
+        Ok(())
+    }
+
+    async fn engine_run<DB>(&self, _: &DB, _: &BootstrapConfig) -> Result<(), LocalDbError>
```

Comment:
do you actually need to do this or can you just not do a default implementation for these fns?

Reply:
the engine implementation for bootstrap is different for both client and producer so we can't use default. are you saying this should be a function in env specific structs and not in the trait itself or something else?